ailiuwSoloProject/ailiuWU_CUI_Project/Market.js

```diff
@@ -0,0 +1,246 @@
+				switch (intent) {
+					case 'Discover Apartments':
+						let department=response.result.parameters.Department;
+						//print products in the department
+						console.log('Here are what we find at the '+department+' department:')
+						let productJson=jsonContent["entities"]
+						console.log(sprintf("%-20s%-10s%-10s", 'Product','Unit','UnitPrice'))						
+						for(let i=0;i<productJson.length;i++){
+							if(productJson[i]["department"]==department){
+								console.log(sprintf("%-20s%-10s%-10f",productJson[i]["value"],productJson[i]["unit"],productJson[i]["price"]))
+							}
+						}
+						console.log('')
+						return askForIntention().then(customerIntents)
+					case 'Add Product':						
+						//store products into the cart
+						let product=response.result.parameters.Product
+						let productNum=parseInt(response.result.parameters.number)
+						let productUnit=''
+						let productPrice=0.0
+						//check if all value are get
+						if((typeof product==='string')&&(productNum>0)){
+							console.log(response.result.fulfillment.speech)
+							console.log('')
+						}
+						else{
+							console.log('Sorry. We can\'t identify what you want')
+							return dialog.question('Could you please re-enter what you want to add to cart')
+								.then(customerIntents)
+						}
+						//add product to the cart
+						if(cartProduct.indexOf(product)<0){
+							let productJson=jsonContent["entities"]
+							for(let i=0;i<productJson.length;i++){
+								if(productJson[i]["value"]==product){
+									//get product price and unit from the stored json data
+									productPrice=productJson[i]["price"]
+									productUnit=productJson[i]["unit"]
+									cartProductUnit.push(productUnit)
+									cartProductPrice.push(productPrice)
+									//console.log(productPrice+productUnit)
+								}
+							}
+							//console.log(jsonContent["entities"][0])
+							cartProduct.push(product)
+							cartProductNum.push(productNum)
+						}
+						else{
+							//add up product number
+							cartProductNum[cartProduct.indexOf(product)]+=productNum
+						}
+						//console.log(product+productNum)
+						//ask for check out or continue shopping
+						return dialog.question({
+									question: 'Would you like to view your cart?',
+									defaultValue: 'no'
+								}).then(viewCart)
+						
+					case 'Get Instructions':
+						//print instructions now
+						instructions()
+						return askForIntention().then(customerIntents)
+					case 'Check Out':
+						return viewCart('yes')
+					case 'View cart':
+						return viewCart('yes')
+					default:
+						return dialog.question(`I don't understand '`+answer+`'. Please re-enter what would you like to do next`)
+									.then(customerIntents)
+				resolve()
+				
+				}
+				
+				
+			})
+						 
+			request.on('error', function(error) {
+				console.log(error)
+				reject()
+			})
+					
+			request.end()
+		})
+}
+
+/*view shopping cart*/
+const viewCart=(answer)=>{
```

Comment:
I like how you broke the intents into separate functions. I think it would make sense to move the addToCart logic into it's own function, to be consistent with this design choice.